### DISCUSSION TITLE
RFProtocol for Dooya 1602 remote

### DISCUSSION THREAD

**Discussion by nosaj66au** [arendst/Tasmota#13055] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
Try to use a DC1602 Dooya Remote with RFSwitch.  I have read the other cases on Dooya Remotes and I believe the 1603 is very similar. To confirm I have compiled the current version with the inclusion of RFRemote. I have tested 2 other remotes (not Dooya), which work correctly and display the received command.

My Question: I have researched the 1602 and it is also a 40bit protocol similar to 1603 (RC protocol 35 or 33) don't revieve work with 40 Bits on the 1603? Do I need to enable 40 reception ? I understand default is 24.      

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ yes] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ yes] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ yes] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ yes] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ yes] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ yes] Device used (e.g., Sonoff Basic): Generic
- [ yes] Tasmota binary firmware version number used: 0x09050007
  - [ ] Pre-compiled
  - [ yes] Self-compiled
- [ ] Flashing tools used: PlatformIO python
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
Not relevant

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
Not relevant

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
Not relevant

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
Not relevant

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Not relevant

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
return an RF message to the console

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._
This demonstrates the 2 working remotes, however the 1602 does not return anything
![image](https://user-images.githubusercontent.com/13111932/132083791-829ddc93-44a0-44c8-bcc3-9d27fb90875f.png)

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment, marked as answer]
If nothing is seen in the console it does mean the protocol is not supported. All known protocols are enabled by default. You can do nothing more here.
So. First the protocol of the Dooya 1602 needs to be known. Seems solved https://community.home-assistant.io/t/dooya-roller-shutters-rf-433mhz/70690/4 With this "someone" can implement in the driver

**Reply by nosaj66au** [reply]
Thanks Jason  however I’m not sure you understand my question.  So I will ask it a little differently.  I understand the 1602 is very similar,  if not the same as the 1603 which is already supported.  I’m simply wanting to know what to expect from a support protocol? My 40 bit question relates to an earlier post about the 1603 protocol which is 40 bit protocol . The RFSwitch is by default 24 bit however the ability to support 40 bits was added November 2020.  Do I need to set a variable or flag to recognise  40 bits or is it supported by default.
Thanks again 
Jason

**Reply by Jason2866** [reply]
It does not count if it is a 24 or 40 bit protocol. Nothing needs to be done if it is supported for all kind of protocols.

**Comment by nosaj66au** [comment]
Thank again Jason so It sounds like is just a matter of timing. I’ll check it out with my DSC tonight and compare the traces between the remote and 1603 protocol output . Should not be that hard to tweak timing. I also have an arduino sketch running as a remote for an additional reference.
Thanks again Jason

**Comment by nosaj66au** [comment]
Hi Again,
   Ok, I've got Tasmota sending commands using protocol 35 to my blinds, and as suspected (protocol 35)  for the DC1603 (5 blinds) and DC1602(15 blinds), are the same, this could be a good reference for others. 
Hope this is my final question on the topic. It appears the RF Receive works on the other remotes I have tried, but not on protocol 35 do we know why this is? I'm not getting console feedback at all. The reason I would like to know, it will allow me to sniff the current remotes and use Tasmota to mimic them, so they will both be able to control the blinds and I don't really wish to program all my blinds again. :)

Thanks Again 
Jason

**Reply by Jason2866** [reply]
Please test #13076 from @barbudor
Compiled firmware with the PR [tasmota-sensors.bin.gz](https://github.com/arendst/Tasmota/files/7115939/tasmota-sensors.bin.gz)

**Reply by barbudor** [reply]
If you don't know how to compile my branch, here are the binaries (bin and bin.gz)
[tasmota-sensors-test.zip](https://github.com/arendst/Tasmota/files/7115944/tasmota-sensors-test.zip)

**Reply by Jason2866** [reply]
Jinx
